packages/datetime/src/timePicker.tsx

```diff
@@ -208,6 +202,12 @@ export class TimePicker extends React.Component<ITimePickerProps, ITimePickerSta
         });
     }
 
+    private handleFocus = (e: React.SyntheticEvent<HTMLInputElement>) => {
+        if (this.props.selectAllOnFocus) {
+            (e.currentTarget as HTMLInputElement).select();
```

Comment:
shouldn't need to cast again here, this is what `SyntheticEvent<T>` is for. 
`currentTarget` is `EventTarget & T`.